eZ/Publish/Core/Repository/UserService.php

```diff
@@ -1294,18 +1294,21 @@ public function validatePassword(string $password, PasswordValidationContext $co
     protected function buildDomainUserGroupObject(APIContent $content)
     {
         $locationService = $this->repository->getLocationService();
-
+        $parentId = null;
         if ($content->getVersionInfo()->getContentInfo()->mainLocationId !== null) {
             $mainLocation = $locationService->loadLocation(
                 $content->getVersionInfo()->getContentInfo()->mainLocationId
             );
-            $parentLocation = $locationService->loadLocation($mainLocation->parentLocationId);
+            try {
+                $parentId = $locationService->loadLocation($mainLocation->parentLocationId)->contentId;
+            } catch (APIUnauthorizedException $e) {
```

Comment:
It's usually not a good idea to call one API service inside another API service just to load extra data. This is why we have persistence layer. What you're trying to do here with heavy API service and exception driven logic can be achieved by permission-less persistence handler call.